torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3453,17 +3453,17 @@ struct to_ir {
       args.emplace_back(loc, "end", end);
     }
     if (sliceable->type()->cast<TupleType>()) {
-      if (step) {
-        // TODO: add support for slicing tuples with a step
-        throw ErrorReport(loc)
-            << "Unsupported operation: slicing tuples with a step isn't supported";
-      }
+      std::vector<at::optional<NamedValue>> tuple_args;
+      tuple_args.reserve(3);
 
-      if (end) {
-        return emitTupleSlice(loc, args[0], args[1], /*end*/ args[2]);
-      } else {
-        return emitTupleSlice(loc, args[0], args[1], c10::nullopt);
-      }
+      start ? tuple_args.emplace_back(start)
```

Comment:
Seems like both options need to be same type for this to work. I tried this before and was getting this "error: operands to ?: have different types ‘torch::jit::Value*’ and ‘const c10::nullopt_t’". I feel like fixing this would make the logic little bit more complicated than necessary.